unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Transform/Utils/ECSTransformUtils.cs

```diff
@@ -48,6 +48,33 @@ public static bool TrySetParent(IParcelScene scene, IDCLEntity child, long paren
         return false;
     }
 
+    public static bool IsCircularParenting(IParcelScene scene, IDCLEntity entity, long parentId)
+    {
+        if (parentId == SpecialEntityId.SCENE_ROOT_ENTITY)
+        {
+            return false;
+        }
+
+        if (parentId == entity.entityId)
+        {
+            return true;
+        }
+
+        do
+        {
+            if (!scene.entities.TryGetValue(parentId, out IDCLEntity parent))
+                break;
+
+            if (entity.entityId == parentId)
+            {
+                return true;
+            }
+            parentId = parent.parentId;
+        } while (parentId != SpecialEntityId.SCENE_ROOT_ENTITY);
```

Comment:
I think all this logic can probably be removed

Reply:
In the end I think we need something to cover the case of the target entity trying to be parented to a child entity of its child hierarchy, but this `do while` solution looks more like it's checking the parents instead of the children

Reply:
OK I just understood how this works: it checks the target parent entity and it traverses from that entiti upwards in the hierarchy checking if the current entity o be parented is in those.